cmd/remoterun/test.go

```diff
@@ -86,16 +82,35 @@ commands:
 				return fmt.Errorf("could not read information for tests: %w", err)
 			}
 
+			kubeClient, _, err := okteto.NewK8sClientProviderWithLogger(k8sLogger).ProvideWithLogger(okteto.GetContext().Cfg, k8sLogger)
+			if err != nil {
+				return fmt.Errorf("could not create kubernetes client: %s", err.Error())
+			}
+
 			// Set the default values for the external resources environment variables (endpoints)
 			for name, external := range dep.External {
 				external.SetDefaults(name)
 			}
 
 			runner := &deployable.TestRunner{
 				Executor: executor.NewExecutor(oktetoLog.GetOutputFormat(), false, ""),
-			}
-			if err != nil {
-				return fmt.Errorf("could not initialize the command properly: %w", err)
+				Fs:       afero.NewOsFs(),
+				GetDevEnvEnviron: func(devEnvName, namespace string) (map[string]string, error) {
+					if devEnvName == "" {
+						return nil, nil
+					}
+					base64Json, err := pipeline.GetConfigmapDependencyEnv(ctx, devEnvName, namespace, kubeClient)
+					if err != nil {
+						return nil, err
+					}
+					return decodeBase64JSON(base64Json)
+				},
+				SetDevEnvEnviron: func(devEnvName, namespace string, vars []string) (err error) {
+					if devEnvName != "" {
+						err = pipeline.UpdateEnvs(ctx, devEnvName, namespace, vars, kubeClient)
```

Comment:
I think we should use other function or at the end, a different (maybe a new one) within the config map, mainly for 2 reasons:
* `OKTETO_ENV` variables are different that dependency envs
* Variables added within `dependencyEnvs` key are, in other parts of the code, added as `OKTETO_DEPENDENCY_<SERVICENAME>_VARIABLE_<VARIABLE>`, but these shouldn't be included. They are not variables belonging to any dependency.

Using the same configmap field to store 2 different things might be misleading, I would create a new field called `oktetoEnv` or similar specifically for these vars. Almost all the logic would be equal, the only thing that would change would be the function to be called here and within `GetDevEnvEnviron`

Reply:
This is not doing anything new. Right now in deploy if you write to `OKTETO_ENV` it will be added to `dependencyEnvs`. I agree it's misleading but I don't think it's something we need to change in this PR. I actually started reading `variables` but then realized they were written to the other key 🤷‍♂️

Reply:
> This is not doing anything new. Right now in deploy if you write to OKTETO_ENV it will be added to dependencyEnvs

Boh, I have been doings tests that is not working fine:
* If the deploy is executed remotely, env vars (`dependencyEnvs`) are not being added to the configmap. That is added through the function `UpdateEnvsFromCommands` but the struct implementing that function for remote executions is `deployInsideDeployConfigMapHandler` and that implementation is empty.
* When redeploying things from the UI, the `dependencyEnvs` is lost between executions. The translations of the configmap happening in the backend is not taking into account the existence of `dependencyEnvs` key, so as soon as it redeploys the dev environment, previous value is lost. I don't know if that is how it should work or a redeploy should be able to read previous okteto envs.

Also, conceptually seems wrong that we use `dependencyEnvs` key to store things to be consumed by the dev environment itself. The idea of that is to be read by a dev environment that has a dependency defined to me. I think that okteto env should be implemented in another key, and both (`dependencyEnvs` and `oktetoEnv`) should be read by the service that has a dependency with the other.

All this is out of the scope of this PR, but we need to review it. It is not a consistent behavior and might lead to odd scenarios.

Reply:
Cool, thanks for the testing. Lets discuss tomorrow in daily. I'll hold on this merging this for now